app/controllers/ViewCollectionsController.scala

```diff
@@ -1,52 +1,42 @@
 package controllers
 
 import akka.http.scaladsl.model.headers.RawHeader
-import graphQL.TDRGraphQLClient
-import io.circe.Decoder
-import io.circe.generic.semiauto.deriveDecoder
+import graphql.GraphQLClientProvider
+import graphql.codegen.GetConsignments
 import javax.inject.{Inject, _}
-import model.TdrCollection
 import modules.TDRAttributes
 import play.api.Configuration
 import play.api.mvc._
-import sangria.macros._
 
 import scala.concurrent.ExecutionContext
 
 @Singleton
 class ViewCollectionsController @Inject()(
-  client: TDRGraphQLClient,
-  controllerComponents: ControllerComponents,
-  configuration: Configuration)(
-  implicit val ex: ExecutionContext) extends AbstractController(controllerComponents) {
+                                           client: GraphQLClientProvider,
+                                           controllerComponents: ControllerComponents,
+                                           configuration: Configuration)(
+                                           implicit val ex: ExecutionContext) extends AbstractController(controllerComponents) {
 
   def index() = Action.async { implicit request: Request[AnyContent] =>
-    case class GetCollectionsQuery(collections: Seq[TdrCollection])
-    case class CollectionsResult(getCollections: GetCollectionsQuery)
-
-    implicit val tdrCollectionDecoder: Decoder[TdrCollection] = deriveDecoder
-    implicit val getCollectionsQueryDecoder: Decoder[GetCollectionsQuery] = deriveDecoder
-    implicit val getCollectionsResultDecoder: Decoder[CollectionsResult] = deriveDecoder
 
     val accessToken = request.attrs.get(TDRAttributes.OAuthAccessTokenKey).get.accessToken
     val header = RawHeader("Authorization", accessToken)
 
-    val appSyncClient = client.appSyncClient(List(header))
-
-    val getCollectionsDoc =
-      graphql"""
-           query MyQuery {
-               getCollections {
-                    collections {
-                      id, name, legalStatus, closure, copyright
-                   }
-                }
-           }
-           """
-
-    appSyncClient.query[CollectionsResult](getCollectionsDoc).result.map(result => result match {
-      case Right(r) => Ok(views.html.showCollections(r.data.getCollections.collections))
+    val appSyncClient = client.graphqlClient(List(header))
+
+    //        appSyncClient.query[CreateConsignment.createCon.Data, CreateConsignment.createCon.Variables](CreateConsignment.createCon.document,
```

Comment:
Do we need this commented out code

Reply:
Left for example of mutation and drunk. Can be removed once a real mutation used